Fallback wayland cursor theme to XCURSOR_THEME

guillaumeboehm opened the pull request:
If the wayland environment doesn't set the cursor properly but the XCURSOR_THEME variable is set, it should probably fallback to it.

lbonn commented on source/wayland/display.c:
```diff
@@ -1210,8 +1210,19 @@ static gboolean wayland_cursor_reload_theme(guint scale) {
   }
   cursor_size *= scale;
 
+  char *cursor_name;
```
`cursor_name` can en up being unitialized here if wayland one is NULL and `XCURSOR_THEME` is not defined.

With this simpler style, there is no need for else branch.

```suggestion
  char *cursor_name = wayland->cursor.theme_name
  if (cursor_name == NULL) {
      // XCURSOR fallback here...
  }
```

guillaumeboehm replied:
Yep thanks, it's amended.

guillaumeboehm committed:
Fallback wayland cursor theme to XCURSOR_THEME

If the wayland environment doesn't set the cursor properly but the
XCURSOR_THEME variable is set, it should fallback to it.

alebastr commented:
Nothing else sets or uses `wayland->cursor.theme_name`, so IMO the patch could be simplified to 
```c
wayland->cursor.theme_name = (char *)g_getenv("XCURSOR_THEME");
```
somewhere in the `wayland_display_setup`
